gems/pending/VolumeManager/MiqLvm.rb

```diff
@@ -312,8 +314,25 @@ def initialize(vgId = nil, vgName = nil, extentSize = nil, seqNo = nil)
   end
 
   def getLvs
-    lvList = []
+    lvList  = []
+    skipLvs = []
     @logicalVolumes.each_value do |lvObj|
+      # 1297019 remove volume groups containing logical volumes w/ 'thin-pool' segments,
+      #         as well as the volumes references in those segments' 'metadata' and 'pool' entries'
+      unless lvObj.thin_segments.empty?
+        skipLvs << lvObj.lvName unless skipLvs.include?(lvObj.lvName)
+        metadata_volumes = lvObj.thin_segments.collect { |segment| segment.metadata }
+        pool_volumes     = lvObj.thin_segments.collect { |segment| segment.pool     }
+        (metadata_volumes + pool_volumes).each do |vol|
+          skipLvs <<  vol unless skipLvs.include?(vol)
+        end
+      end
+
+      if skipLvs.include?(lvObj.lvName)
+        $log.debug "Ignoring thin volume: #{lvObj.lvName}"
+        next
+      end
```

Comment:
I think this logic might not account for ordering differences. If we encounter a metadata volume or pool volume before we encounter the LV that references them, then they won't be in the `skipLvs` list at this point - so they won't be skipped.

You probably have to do this in 2 passes. First, loop through the LVs building the `skipLvs` list. Then, loop through the LVs again, checking: `skipLvs.include?(lvObj.lvName)` after `skipLvs` has been fully populated.